instrumentation/opentelemetry-instrumentation-requests/src/opentelemetry/instrumentation/requests/__init__.py

```diff
@@ -234,6 +238,9 @@ def get_or_create_headers():
 
         url = redact_url(request.url)
 
+        # Get headers early for user agent detection
+        headers = get_or_create_headers()
```

Comment:
Nit: Do we need to get headers early? Why can't we populate synthetic type when we get it below?

Reply:
Logically makes more sense to keep it close to where we use the headers value. Good callout.